orchestration/dagster_orchestration/hca_orchestration/tests/contrib/test_argo_workflows.py

```diff
@@ -0,0 +1,180 @@
+                'data-repo-name': 'datarepo_dataset1'
+            }),
+            mock_argo_workflow('import-hca-total-cdef', 'abc345uid', 'Succeeded', params={
+                'data-repo-name': 'datarepo_dataset2'
+            }),
+        ]
+
+        archived_workflow_list_3_workflows = self.workflows_to_page_of_results(list_containing_3_V1alpha1Workflows,
+                                                                               "page2")
+
+        list_containing_1_V1alpha1Workflows = [
+            mock_argo_workflow('import-hca-total-page2', 'page2uid', 'Supsneeded', params={
+                'data-repo-name': 'datarepo_dataset1'
+            }),
+        ]
+
+        archived_workflow_list_1_workflows = self.workflows_to_page_of_results(list_containing_1_V1alpha1Workflows,
+                                                                               "page3")
+
+        list_containing_2_V1alpha1Workflows = [
+            mock_argo_workflow('import-hca-1', '12345', 'Succeeded', params={
+                'data-name': 'dataset',
+                'test': 'data'
+            }),
+            mock_argo_workflow('hca-data', 'u123id', 'Succeeded'),
+        ]
+
+        archived_workflow_list_2_workflows = self.workflows_to_page_of_results(list_containing_2_V1alpha1Workflows,
+                                                                               "page4")
+
+        list_containing_0_V1alpha1Workflows = [
+        ]
+
+        archived_workflow_list_0_workflows = self.workflows_to_page_of_results(list_containing_0_V1alpha1Workflows,
+                                                                               None)
+
+        def get_page(list_options_continue=None):
+            {
+                None: archived_workflow_list_3_workflows,
+                "page2": archived_workflow_list_1_workflows,
+                "page3": archived_workflow_list_2_workflows,
+                "page4": archived_workflow_list_0_workflows
+            }[list_options_continue]
+
+            list_results = list(self.client._pull_paginated_results(get_page()))
+            self.assertEqual(list_results,
+                             list_containing_3_V1alpha1Workflows +
+                             list_containing_1_V1alpha1Workflows +
+                             list_containing_2_V1alpha1Workflows +
+                             list_containing_0_V1alpha1Workflows)
+            self.assertEqual(list_results.call_count, 4)
+            list_results.assert_has_calls([call(),
+                                           call(list_options_continue="page2"),
+                                           call(list_options_continue="page3"),
+                                           call(list_options_continue="page4")])
+
+    def test_inflate_one_workflow_one_call(self):
+        v1alphaworkflow_mock = mock_argo_workflow('import-hca-total-abcd', 'abc234uid', 'Succeeded', params={
```

Comment:
here you can use a name like `mock_inflated_workflow`, since it's going to be returned by the inflation attempt.